docs/CORE_DEV_GUIDE.md

```diff
@@ -0,0 +1,180 @@
+consult back with our [mission and values](MISSION_AND_VALUES.md).
+
+2. **Performance and benchmarks:** As `napari` targets scientific applications which often involve
+large multidimensional datasets, high performance is a key value of `napari`. While
+every new feature won't scale equally to all sizes of data keeping in mind performance
+and our [benchmarks](BENCHMARKS.md) during a review may be important, and you may
+need to ask for benchmarks to be run and reported or new benchmarks to be added.
+
+3. **APIs and stability:** Coding users and plugin developers will make
+extensive use of our APIs. The foundation of a healthy plugin ecosystem will be
+a fully capable and stable set of APIs and so as the `napari` matures it will
+very important to ensure our APIs are stable. For now, while the project is still
+in an earlier stage spending the extra time to consider names of public facing
+variables and methods, along side function signatures, could save us considerable
+trouble in the future.
+
+4. **Documentation and tutorials:** All new methods should have appropriate doc
+strings following [PEP257](https://www.python.org/dev/peps/pep-0257/) and the
+[NumPy documentation guide](https://docs.scipy.org/doc/numpy/docs/howto_document.html).
+For any major new features accompanying changes should be made to our
+[tutorials repository](https://github.com/napari/napari-tutorials), that not only
+illustrates the new feature, but explains it.
+
+5. **Implementations and algorithms:** You should understand the code being modified
+or added before approving it.  (See `Merge Only Changes You Understand` below.)
+Implementations should do what they claim, and be simple, readable, and efficient.
+
+6. **Tests:** All contributions *must* be tested, and each added line of code
+should be covered by at least one test. Good tests not only execute the code,
+but explores corner cases.  It can be tempting not to review tests, but please
+do so.
+
+Other changes may be *nitpicky*: spelling mistakes, formatting,
+etc. Do not insist contributors make these changes, and instead you can
+make the changes by [pushing to their branch](https://help.github.com/articles/committing-changes-to-a-pull-request-branch-created-from-a-fork/), or using GitHub’s [suggestion](https://help.github.com/articles/commenting-on-a-pull-request/) [feature]
+(https://help.github.com/articles/incorporating-feedback-in-your-pull-request/).
+(The latter is preferred because it gives the contributor a choice in
+whether to accept the changes.)
+
+Unless you know that a contributor is experienced with git, don’t
+ask for a rebase when merge conflicts arise. Instead, rebase the
+branch yourself, force-push to their branch, and advise the contributor to force-pull.  If the contributor is
+no longer active, you may take over their branch by submitting a new pull
+request and closing the original. In doing so, ensure you communicate
+that you are not throwing the contributor's work away!
```

Comment:
and include a reference to the original pull request

Reply:
done